### DISCUSSION TITLE
Uploading collection via Irys with Account

### DISCUSSION THREAD

**Discussion by CryptainGolD** [aptos-labs/aptos-developer-discussions#506] [Questions] [answered]
### Describe your question in detail.

How to upload collection via Irys with account keypair in NFT-Minting-Dapp template? WalletContextState does not needed because this project is for telegram bot

I've defined Irys.ts as follows:
```
import { NodeIrys } from "@irys/sdk";
import { getAccountAPTBalance } from "../view-functions/getAccountAPTBalance";
import { NETWORK } from "../config";
const getRpcUrl = (network: string): string => {
  switch (network) {
    case "mainnet":
      return "https://fullnode.mainnet.aptoslabs.com";
    case "testnet":
      return "https://fullnode.testnet.aptoslabs.com";
    default:
      throw new Error(`Unsupported network: ${network}`);
  }
};
const getNodeIrys = async (account: any) => {
  const token = "aptos";
  const rpcUrl = getRpcUrl(NETWORK);
  const nodeIrys = await NodeIrys.init({
    url: rpcUrl,
    token,
    privateKey: account.privateKey,
    publicKey: account.accountAddress,
  });
  await nodeIrys.ready();
  return nodeIrys;
};
```

`account` is correct as Account from ts-sdk
```
export const getAccountFromPriKey = async (
  args: GetAccountFromPriKeyArguments
): Promise<any> => {
  const { privateKeyByte } = args;
  const privateKey = new Ed25519PrivateKey(privateKeyByte);

  const account = Account.fromPrivateKey({ privateKey });
  return account;
};
```

### What error, if any, are you getting?

`TypeError: Cannot read properties of undefined (reading 'publicKey')`

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0x-j** [comment]
```
  const nodeIrys = await NodeIrys.init({
    url: rpcUrl,
    token,
    privateKey: account.privateKey,
    // you are using account.address, it should be account.publicKey
    publicKey: account.publicKey,
  });
```

**Comment by 0xmaayan** [comment, marked as answer]
You can check Irys official docs at https://docs.irys.xyz/build/d/sdk/setup#aptos or if you are on a web environment https://docs.irys.xyz/build/d/irys-in-the-browser#aptos

Also note that Irys does not support Aptos devnet at the moment, so your program should only support Aptos mainnet/testnet

**Reply by CryptainGolD** [reply]
Thanks for your reply. I've resolved it to follow docs, already.
